airflow/contrib/operators/gcs_download_operator.py

```diff
@@ -32,6 +32,10 @@
 class GoogleCloudStorageDownloadOperator(BaseOperator):
     """
     Downloads a file from Google Cloud Storage.
+    When no filename is supplied, the operator loads the file into memory and returns its
```

Comment:
This isn't quite true for an operator as operators don't have a return value, so it's the store_to_xcom_key that controls the behaviour.

As a result of your change in the hook `store_to_xcom_key` and `filename` are now mutually exclusive, which makes this a breaking change to the operator behaviour. 

So either you need to add a note to UPDATING.md about this change (and ensure that either store_to_xcom_key is True and filename is empty) or make the operator load the contents form disk when store_to_xcom_key is True and filename is not empty.

Reply:
@ashb Thank you for your comment, I've changed the behaviour, updated the docstring and included the info into the UPDATING.md